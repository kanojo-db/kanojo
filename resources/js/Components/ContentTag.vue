<script setup lang="ts">
import type { Tag } from '@/types/models';
import type { PropType } from 'vue';

import { useI18n } from 'vue-i18n';

import { useName } from '@/utils/item';

const props = defineProps({
    tag: {
        type: Object as PropType<Tag>,
        required: true,
    },
});

const locale = useI18n().locale.value;

const name = useName(props.tag, locale);
</script>

<template>
    <v-chip>{{ name }}</v-chip>
</template>
