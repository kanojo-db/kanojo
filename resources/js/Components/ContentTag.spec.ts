import type { Tag } from '@/types/models';

import { renderToString, shallowMount } from '@vue/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

import ContentTag from '@/Components/ContentTag.vue';
import getI18nPlugin from '@/plugins/i18n';
import getVuetifyPlugin from '@/plugins/vuetify';

expect.extend(toHaveNoViolations);

const i18n = getI18nPlugin('en-US');
const vuetify = getVuetifyPlugin(i18n);

const tag: Tag = {
    id: 1,
    name: {
        'en-US': 'Test Tag',
        'ja-JP': 'テストタグ',
    },
    type: 'type',
    slug: 'test-tag',
    order_column: 1,
    created_at: '2020-01-01 00:00:00',
    updated_at: '2020-01-01 00:00:00',
};

describe('ContentTag', () => {
    it('renders properly in SSR mode', async () => {
        const contents = await renderToString(ContentTag, {
            propsData: {
                tag,
            },
            global: {
                plugins: [i18n, vuetify],
            },
        });

        expect(contents).toMatchSnapshot();
    });

    it('has no a11y violations', async () => {
        const wrapper = shallowMount(ContentTag, {
            propsData: {
                tag,
            },
            global: {
                plugins: [i18n, vuetify],
            },
        });

        const results = await axe(wrapper.html(), {
            rules: {
                region: { enabled: false },
            },
        });

        expect(results).toHaveNoViolations();
    });
});
