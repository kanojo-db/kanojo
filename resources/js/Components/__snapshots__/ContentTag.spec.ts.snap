// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentTag > renders properly in SSR mode 1`] = `"<span class=\\"v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal\\" style=\\"\\" draggable=\\"false\\"><!--[--><!----><span class=\\"v-chip__underlay\\"></span><!--]--><!----><!----><div class=\\"v-chip__content\\"><!--[-->Test Tag<!--]--></div><!----><!----></span>"`;
